### ISSUE TITLE
PreBuildEvent task changes to Directory.Build.targets doesn't affect the build

### ISSUE THREAD

**Issue by SidShetye** [dotnet/sdk#10941]
We're using a `PreBuildEvent` target to run a script `InjectVersionToBuildProps.ps1` that updates the `Directory.Build.targets` (solution root) with a `Version` and `InformationalVersion`.

```
<Project Sdk="Microsoft.NET.Sdk.Web">
  <!-- snipped -->
  <Target Name="PreBuild" BeforeTargets="PreBuildEvent">
    <Exec Command="pwsh -ExecutionPolicy ByPass -File  $(SolutionDir)scripts\build\InjectVersionToBuildProps.ps1" />    
  </Target>
</Project>
```

The powershell script itself successfully updates `Directory.Build.targets` with the latest values. 

The issue is that the build still ends up with the *previous* values from `Directory.Build.targets` - _not_ the just-updated from our script in the `PreBuildEvent`. 

In other words,

#### Directory.Build.targets before running the build
```
<Project>
  <PropertyGroup>
    <Version>1.0.0.0</Version>
    <InformationalVersion>PlaceHolder</InformationalVersion>
  </PropertyGroup>
</Project>
```

#### Directory.Build.targets after running the build
```
<Project>
  <PropertyGroup>
    <Version>2020.1.07.0815</Version>
    <InformationalVersion>server-enterprise/20200107_081500-1-g1fa18dba</InformationalVersion>
  </PropertyGroup>
</Project>
```

The build still ends up with `Version=1.0.0.0` and `InformationalVersion=PlaceHolder`. 

With a self-documenting name like `PreBuildEvent`, this behavior is wrong, hence reporting it here.

**Comment by wli3** [comment]
msbuild has multiple pass. Property is evaluated in the first pass before running other Targets. So it cannot be changed from the file after the Target is run. This scenario is not supported

**Comment by SidShetye** [comment]
> This scenario is not supported

@wli3 Then what's a supported `msbuild` way to update the `Directory.Build.targets` with a powershell script?
